src/serial.rs

```diff
@@ -1047,27 +1130,26 @@ macro_rules! hal {
                 }
             }
 
-            impl Tx<$USARTX> {
+            impl $txdma {
                 /// Creates a new DMA frame sender
                 pub fn frame_sender<BUFFER, N>(
-                    &self,
-                    mut channel: $tx_chan,
-                ) -> FrameSender<BUFFER, $tx_chan, N>
+                    mut self,
+                ) -> FrameSender<BUFFER, Self, N>
                     where
                         BUFFER: Sized + StableDeref<Target = DMAFrame<N>> + DerefMut + 'static,
                         N: ArrayLength<MaybeUninit<u8>>,
                 {
                     let usart = unsafe{ &(*$USARTX::ptr()) };
 
                     // Setup DMA
-                    channel.set_peripheral_address(&usart.tdr as *const _ as u32, false);
+                    self.channel.set_peripheral_address(&usart.tdr as *const _ as u32, false);
 
                     // Tell DMA to request from serial
-                    channel.cselr().modify(|_, w| {
-                        w.$dmacst().bits(0b0010) // TODO: Fix this, not valid for DMA2
```

Comment:
According to [this](https://www.st.com/content/ccc/resource/technical/document/reference_manual/02/35/09/0c/4f/f7/40/03/DM00083560.pdf/files/DM00083560.pdf/jcr:content/translations/en.DM00083560.pdf#%5B%7B%22num%22%3A535%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C67%2C584%2Cnull%5D) it seems to always be map2